doc/collision_prediction_report.Rmd

```diff
@@ -0,0 +1,46 @@
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+library(knitr)
+library(kableExtra)
+library(tidyverse)
+```
+
+```{r load model results}
+final_model_scores <- read_csv("../results/scores_test2.csv", show_col_types = FALSE)
+```
+
+# Introduction
+
+Stemming from the daily news we come across on fatalities due to motor vehicle collisions, we thought about designing a model that would help predict whether a motor vehicle collision would result in a fatality or not. This lead to our predictive research question: "Will a motor vehicle collision result in fatalities?". But would it suffice to only "predict" a fatality? What if we also determine some of the major contributors that could lead to a fatal collision? Through this predictive analysis we first aim to predict whether a collision would result in a fatality or not followed by determining what features are most important in making these predictions. Specifically, we want to know how important month, roadway configuration, weather condition, person age, and vehicle type are in predicting the severity of a motor vehicle collision.
+
+# Analysis
+## Data
+The data set that will be used in this project has been sourced from the National Collision Database, published by Transport Canada, which can be found [here](https://open.canada.ca/data/en/dataset/1eb9eba7-71d1-4b30-9fb1-30cbdab7e63a). The National Collision Database contains data on all of the police-reported motor vehicle collisions on public roads in Canada from 1999 to the most recent available data from 2017. We will start off our analysis using the data collected from collisions that occurred in 2017. This data set contains information licensed under the Open Government Licence – Canada.
+
+## Model
+The Logistic Regression algorithm was used to build a classification model to predict whether a motor vehicle collision lead to a fatality or not. All variables included in the original data set, with the exception of "C_YEAR", "C_CASE", "C_SEV" were used to fit the model. The "P_ISEV" column served as the target column to be predicted. The hyperparameter's $C$ and $Gamma$ were chosen using 5-fold cross validation with random search `RandomizedSearchCV`. In order to tackle the issue of class imbalance, we performed random undersampling using `RandomUnderSampler`. This helped us in determining the Logistic Regression model with the best parameters. We further performed feature selection using `RandomUnderSampler`, `OneHotEncoder`, `RFECV`, and `LogisticRegression` model to reduce the number of features. The R and Python programming languages [@R; @Python] and the following R and Python packages were used to perform the analysis: knitr [@knitr], docopt [@docoptpython], os [@Python], Pandas [@mckinney-proc-scipy-2010], scikit-learn [@scikit-learn], Altair [@altair], Vegalite [@vegalite]. The code used to perform the analysis and create this report can be found here: <https://github.com/UBC-MDS/Collision_Prediction>.
```

Comment:
The Logistic Regression algorithm was used to build a classification model to predict whether a motor vehicle collision leads to a fatality or not. All variables included in the original data set, with the exception of "C_YEAR", "C_CASE", "C_SEV", "P_ISEV", "V_ID", and "P_ID" were used to fit the model, as these features were either irrelevant to our prediction or contained redundant information. The "FATALITY" column was created using the "P_ISEV" by converting values that correspond to a fatality as "True", and  converting values that correspond to an injury or no injury as "False". The "FATALITY" column served as the target column to be predicted. In order to tackle the issue of class imbalance and to perform cross-validations in feasible times with our available computing power, we performed random undersampling using `RandomUnderSampler`. Since all features were categorical, our pipeline consisted of the `RandomUnderSampler `, a `OneHotEncoder`, and the logistic regression model. We determined the optimal value of the logistic regression model's  hyperparameter, $C$, using 5-fold cross validation with random search `RandomizedSearchCV`.  We further performed feature selection using `RandomUnderSampler`, `OneHotEncoder`, `RFECV`, and `LogisticRegression` model to reduce the number of features. The R and Python programming languages [@R; @Python] and the following R and Python packages were used to perform the analysis: knitr [@knitr], docopt [@docoptpython], os [@Python], Pandas [@mckinney-proc-scipy-2010], scikit-learn [@scikit-learn], Altair [@altair], Vegalite [@vegalite]. The code used to perform the analysis and create this report can be found here: <https://github.com/UBC-MDS/Collision_Prediction>.